.circleci/config.yml

```diff
@@ -216,62 +190,43 @@ jobs:
           command: echo "$ARTIFACTORY_SETTINGS" | base64 --decode > artifactory-settings.xml
       - run:
           name: Deploy binaries to artifactory
+          no_output_timeout: 120m
           command: |
             source $BASH_ENV
             ./boost.sh --no-clean --no-unpack --no-framework --deploy `test -n "${BOOST_VERSION:-}" && echo --boost-version $BOOST_VERSION` \
-              `test -n "${TWILIO_SUFFIX:-}" && echo --twilio-suffix -$TWILIO_SUFFIX` --boost-libs "<< parameters.boost-libs >>"
-          no_output_timeout: 120m
+              `test -n "${TWILIO_SUFFIX:-}" && echo --twilio-suffix -$TWILIO_SUFFIX` --boost-libs "<< pipeline.parameters.boost-libs >>"
 
 workflows:
-  version: 2
   build-and-deploy:
     jobs:
       - unpack-sources:
           <<: *ignore-master
           name: unpack
           context: dockerhub-pulls
-      - build-headers:
-          <<: *ignore-master
-          name: build-boost-headers
-          context: dockerhub-pulls
-          requires:
-            - unpack
-      - build-android:
-          <<: *ignore-master
-          name: build-boost-android
-          context: dockerhub-pulls
-          requires:
-            - unpack
-      - build-linux:
-          <<: *ignore-master
-          name: build-boost-linux
-          context: dockerhub-pulls
-          requires:
-            - unpack
-      - build-linux-cxx11-abi-disabled:
+
+      - build-on-linux:
           <<: *ignore-master
-          name: build-boost-linux-cxx11-abi-disabled
+          name: build-boost-<< matrix.platform >>
           context: dockerhub-pulls
           requires:
             - unpack
-      - build-ios:
-          <<: *ignore-master
-          name: build-boost-ios
-          requires:
-            - unpack
-      - build-osx:
+          matrix:
+            parameters:
+              platform: [ headers, linux, linux-cxx11-abi-disabled, android ]
+
+      - build-on-mac:
           <<: *ignore-master
-          name: build-boost-osx
+          name: build-boost-<< matrix.platform >>
           requires:
             - unpack
+          matrix:
+            parameters:
+              platform: [ ios, osx ]
```

Comment:
So the matrix is 1D? The build step is repeated for each entry in the parameters I assume.

Reply:
One job is created for each parameter combination, and yes, in this case it is a 1D matrix, which means 2 jobs.